inputFiles/singlePhaseFlow/thermalCompressible_2d_base.xml

```diff
@@ -24,21 +24,21 @@
     </FiniteVolume>
   </NumericalMethods>
 
-  <ElementRegions>
-    <CellElementRegion
-      name="region"
-      cellBlocks="{ cb }"
-      materialList="{ fluid, rock, thermalCond }"/>
-  </ElementRegions>
-
   <Constitutive>
 
     <CompressibleSolidConstantPermeability
-      name="rock"
+      name="rock_linear"
+      solidModelName="nullSolid"
+      porosityModelName="rockPorosity"
+      permeabilityModelName="rockPerm"
+      solidInternalEnergyModelName="rockInternalEnergy_linear"/>
+
+    <CompressibleSolidConstantPermeability
+      name="rock_nonLinear"
       solidModelName="nullSolid"
       porosityModelName="rockPorosity"
       permeabilityModelName="rockPerm"
-      solidInternalEnergyModelName="rockInternalEnergy"/>
+      solidInternalEnergyModelName="rockInternalEnergy_nonLinear"/>
```

Comment:
suggest to use the name as `rockInternalEnergy_linear`

Reply:
The volumetric heat capacity depends linearly on temperature then the internal energy is non-linearly related to temperature. Therefore I propose to use the _nonLinear term hear because it describe the non-linear internal energy law.